cares-center-inventory-Song/crud.py

```diff
@@ -0,0 +1,242 @@
+
+def get_transaction_histories_by_manager_id(db: Session, manager_id: str):
+    return db.query(models.TransactionHistory).filter(models.TransactionHistory.manager_id == manager_id).all()
+
+def delete_transaction_history_by_transaction_id(db: Session, transaction_id: str):
+    transaction_history_to_delete = db.query(models.TransactionHistory).filter(models.TransactionHistory.transaction_id == transaction_id).first()
+    db.delete(transaction_history_to_delete)
+    db.commit()
+
+# transaction_items
+def create_transaction_item(db: Session, transaction_item = schemas.TransactionItemCreate):
+    db_transaction_item = models.TransactionItem(
+        item_id = transaction_item.item_id,
+        transaction_id = transaction_item.transaction_id,
+        transaction_quantity = transaction_item.transaction_quantity
+    )
+    db.add(db_transaction_item)
+    db.commit()
+    db.refresh(db_transaction_item)
+
+def get_transaction_items_by_transaction_id(db: Session, transaction_id: str):
+    return db.query(models.TransactionItem).filter(models.TransactionItem.transaction_id == transaction_id).all()
+
+def delete_transaction_items_by_item_id(db: Session, item_id: str):
+    transaction_item_to_delete = db.query(models.TransactionItem).filter(models.TransactionItem.item_id == item_id).all()
+    db.delete(transaction_item_to_delete)
+    db.commit()
+
+def delete_transaction_items_by_transaction_id(db: Session, transaction_id: str):
+    transaction_item_to_delete = db.query(models.TransactionItem).filter(models.TransactionItem.transaction_id == transaction_id).all()
+    db.delete(transaction_item_to_delete)
+    db.commit()
+
+# donor_weight
+def create_donor_weight(db: Session, donor_weight:schemas.DonorWeightCreate):
+    db_donor_weight = models.DonorWeight(
+        donor_id = donor_weight.donor_id,
+        transaction_id = donor_weight.transaction_id,
+        weight = donor_weight.donor_id
+    )
+    db.add(db_donor_weight)
+    db.commit()
+    db.refresh(db_donor_weight)
+
+def get_donor_weights_by_transaction_id(db: Session, transaction_id: str):
+    return db.query(models.DonorWeight).filter(models.DonorWeight.transaction_id == transaction_id).all()
+
+def get_donor_weights_by_donor_id(db: Session, donor_id: str):
+    return db.query(models.DonorWeight).filter(models.DonorWeight.donor_id == donor_id).all()
+
+def get_donor_weights_by_transaction_date(db: Session, transaction_date: str):
+    return db.query(models.DonorWeight).join(models.TransactionHistory).filter(models.TransactionHistory.transaction_date == transaction_date).all()
+
+def get_donor_weights_by_donor_id_and_transaction_date(db: Session, donor_id: str, transaction_date: str):
+    return db.query(models.DonorWeight).join(models.TransactionHistory).filter(and_(models.DonorWeight.donor_id == donor_id, models.TransactionHistory.transaction_date == transaction_date)).all()
+
+def delete_donor_weights_by_donor_id(db: Session, donor_id: str):
```

Comment:
Combine this one with the function starting on line 239 so it takes a donor_id and a transaction_id